### DISCUSSION TITLE
Alternative text for links and references

### DISCUSSION THREAD

**Discussion by ajede** [Mogztter/asciidoctor-web-pdf#660] [Q&A]
Currently I am using Asciidoctor-PDF to generate User documentation files. But we need to make the PDF accessible and conform to PDF/UA standard, so I turned to asciidoctor-web-pdf.
But my problem is that I need to have alternative Texts for links and internal references to other chapters.

Is this possible? I have not found a way to add alternative texts similar to alternative text for images.

**Comment by ggrossetie** [comment]
> But my problem is that I need to have alternative Texts for links and internal references to other chapters.

Could you please provide an example of the AsciiDoc syntax you are using with the current result and the expected result?

**Comment by ajede** [comment]
Dear Guillame,

Since I asked this question in the chat, I have learned some more information.
The Matterhorn Protocol requires an alternate description in the Contents Key of every Link tag, so I did not word it correctly with "alternative text". 
The minimum requirement  and maybe easiest solution would be to set the value of the Contents Key to the title attribute. The value type is always String.

What I do to create links to other chapters in the document is to use custom anchors, such as <<setting>>.  I usually want to have the output "Chapter x," but sometimes I also include some text to display: <<setting,Setting>>. 
To try to use the title attribute as described above, I tried something like this: <<setting, title="User settings for xyz">>, but that was not successfull. 
The contents key is always empty. 

I hope that helps,

Aline



Aline Jede
Technischer Redakteur
Entwicklung
Division Public Authorities
secunet Security Networks AG 

Tel.: +49 201 5454 2753
E-Mail: ***@***.***
Ammonstraße 74, 01067 Dresden
www.secunet.com
______________________________________________________________________

secunet Security Networks AG
Sitz: Kurfürstenstraße 58, 45138 Essen, Deutschland
Amtsgericht Essen HRB 13615
Vorstand: Axel Deininger (Vors.), Torsten Henn, Dr. Kai Martius, Thomas Pleines
Aufsichtsratsvorsitzender: Dr. Ralf Wintergerst
__________________________________________________________________

-----Ursprüngliche Nachricht-----
Von: Guillaume Grossetie ***@***.***> 
Gesendet: Donnerstag, 28. Juli 2022 00:32
An: Mogztter/asciidoctor-web-pdf ***@***.***>
Cc: Jede, Aline ***@***.***>; Author ***@***.***>
Betreff: Re: [Mogztter/asciidoctor-web-pdf] Alternative text for links and references (Discussion #660)

	But my problem is that I need to have alternative Texts for links and internal references to other chapters.

Could you please provide an example of the AsciiDoc syntax you are using with the current result and the expected result?

**Reply by ggrossetie** [reply]
What about `reftext`?
https://docs.asciidoctor.org/asciidoc/latest/macros/xref-text-and-style/#default-styling


```
[reftext="User settings for xyz"]
=== Settings


xref:xyz.adoc#settings[]
```

In this case, the text of the cross reference link appears as: `User settings for xyz`.
We could support an `alt` or `title` on `xref` but currently we do not use this attribute. I also don't know if Chromium PDF engine supports this feature since it generates a PDF from an HTML document and `<a>` does not have an alt or title attribute: https://developer.mozilla.org/en-US/docs/Web/HTML/Element/a

**Comment by ajede** [comment]
Dear Guillaume,

Thanks for your reply.

I was previously using Asciidoctor-pdf and just tried asciidoctor-web-pdf, because I need to create accessible user documentation.

Unfortunately even though the pdf is tagged, the tagging is not really great and sometimes incorrect, so I will not be using asciidoctor-web-pdf at the moment.
I would love to try again, when the major issues are solved:

- title or alt tag for cross references
- bounding box for images
- lists over multiple pages must be tagged in one list element
- lists have a lbl and a lbody tag
- table of contents is tagged correctly
- unnecessary div tags are removed
- text that spans over multiple lines is not tagged with a <p> element for each element

Best regards,
Aline Jede

Aline Jede
Technischer Redakteur
Entwicklung
Division Public Authorities
secunet Security Networks AG 

Tel.: +49 201 5454 2753
E-Mail: ***@***.***
Ammonstraße 74, 01067 Dresden
www.secunet.com
______________________________________________________________________

secunet Security Networks AG
Sitz: Kurfürstenstraße 58, 45138 Essen, Deutschland
Amtsgericht Essen HRB 13615
Vorstand: Axel Deininger (Vors.), Torsten Henn, Dr. Kai Martius, Thomas Pleines
Aufsichtsratsvorsitzender: Dr. Ralf Wintergerst
__________________________________________________________________

-----Ursprüngliche Nachricht-----
Von: Guillaume Grossetie ***@***.***> 
Gesendet: Freitag, 29. Juli 2022 17:35
An: Mogztter/asciidoctor-web-pdf ***@***.***>
Cc: Jede, Aline ***@***.***>; Author ***@***.***>
Betreff: Re: [Mogztter/asciidoctor-web-pdf] Alternative text for links and references (Discussion #660)

What about reftext?
https://docs.asciidoctor.org/asciidoc/latest/macros/xref-text-and-style/#default-styling

[reftext="User settings for xyz"]
=== Settings


xref:xyz.adoc#settings[]


In this case, the text of the cross reference link appears as: User settings for xyz.
We could support an alt or title on xref but currently we do not use this attribute.
